docs/utilities/parameters.md

```diff
@@ -515,3 +515,76 @@ The **`config`** and **`boto3_session`** parameters enable you to pass in a cust
 		value = ssm_provider.get("/my/parameter")
 		...
 	```
+
+## Testing your code
+
+For unit testing your applications, you can mock the calls to the parameters utility to avoid calling AWS APIs. This
+can be achieved in a number of ways - in this example, we use the [pytest monkeypatch fixture](https://docs.pytest.org/en/latest/how-to/monkeypatch.html)
+to patch the `parameters.get_parameter` method:
+
+=== "tests.py"
+	```python
+	from src.index import handler
+
+	def test_handler(monkeypatch):
+
+		def mockreturn(name):
+			return "mock_value"
+
+		monkeypatch.setattr(parameters, "get_parameter", mockreturn)
+		return_val = handler({}, {})
+		assert return_val.get('message') == 'mock_value'
+
+	```
+
+=== "src/index.py"
+	```python
+	from aws_lambda_powertools.utilities import parameters
+
+	def handler(event, context):
+		# Retrieve a single parameter
+		value = parameters.get_parameter("my-parameter-name")
+		return {"message": value}
+	```
+
+If we need to use this pattern across multiple tests, we can avoid repetition by refactoring to use our own pytest fixture:
+
+=== "tests.py"
+	```python
+	import pytest
+
+	from src.index import handler
+
+	@pytest.fixture
+	def mock_parameter_response(monkeypatch):
+		def mockreturn(name):
+			return "mock_value"
+
+    	monkeypatch.setattr(parameters, "get_parameter", mockreturn)
```

Comment:
**Q**: Shall we return a factory so they can control the return value at each test?

e.g. `mock_parameter_response(data_i_want_to_return)`

Reply:
It can be done but its not pretty! This is slightly different to the usual fixture, since we don't actually care about the fixture's value, we just need it to run so it monkeypatches the `get_parameter` method. We _could_ do something like the below example, but IMHO it is a lot more readable to just have multiple fixtures where needed:

```python
@pytest.fixture
def mock_parameter_response(monkeypatch):
    def factory_func(parameter_return_value):
        def mockreturn(name):
            return parameter_return_value
        monkeypatch.setattr(parameters, "get_parameter", mockreturn)

    return factory_func


def test_handler(mock_parameter_response):
    mock_parameter_response("aaaaaaaaargh!")
    return_val = handler({}, {})
    assert return_val.get('message') == "aaaaaaaaargh!"
```

Reply:
ugh yep indeed. Let's keep as-is.